### PR TITLE
C#: Improve getRuntimeArgumentForParameter to consider named arguments.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we improve the predicate `getRuntimeArgumentForParameter` to correctly consider named arguments when calling runtime targets. The implementation mirrors the implementation for static call targets (including the pragmas as they are probably there for a good reason).
That is, `getRuntimeArgumentForParameter` now correctly identifies the provided values for the named arguments `x` and `y` when calling `sum` in the example below.

```csharp
delegate int MyDelegate(int x, int y);

void M1()
{
    MyDelegate sum = (int x, int y) => x + y;
    sum(y: 4, x: 5);
}
```

### CODE REVIEW

**Comment by michaelnebel** [comment]
DCA looks good.

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll [hunk 1]
```diff
@@ -183,24 +183,39 @@ class Call extends DotNet::Call, Expr, @call {
...
+      // Appears in the positional part of the call
+      result = this.getImplicitRuntimeArgument(p)
+      or
+      // Appears in the named part of the call
+      result = this.getExplicitRuntimeArgument(p.getName())
+    )
+  }
+
+  pragma[noinline]
+  private Expr getImplicitRuntimeArgument(Parameter p) {
```
[the remark this answers is not in the document]
Comment (tamasvajk) [reply csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:202]: I think I introduced this problem in `getImplicitArgument` in https://github.com/github/codeql/pull/15175/files. We should probably fix that too.

File: csharp/ql/lib/semmle/code/csharp/exprs/Call.qll [hunk 2]
```diff
@@ -188,7 +189,7 @@ class Call extends DotNet::Call, Expr, @call {
    */
   cached
   Expr getRuntimeArgumentForParameter(Parameter p) {
-    p = this.getARuntimeTarget().getAParameter() and
+    this.getARuntimeTarget().getAParameter() = p and
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:192]: This conjunct can now be moved down to
```
result = this.getExplicitRuntimeArgument(p.getName())
```

Comment (hvitved) [reply csharp/ql/lib/semmle/code/csharp/exprs/Call.qll:192]: The same goes for `getArgumentForParameter`.

**Review by hvitved — CHANGES_REQUESTED** [review]

**Review by tamasvajk — APPROVED** [review]

**Review by hvitved — APPROVED** [review]

**Pull request merged**